`@xyflow/react` fails with pnpm strict mode — missing peerDependency on `@types/react`

`@xyflow/react@12.10.1` ships `.d.ts` files that import from `react`:

```ts
import { ComponentType } from 'react';
```

TypeScript resolves these types through `@types/react`, but `@xyflow/react` only declares `react` and `react-dom` as peer dependencies — not `@types/react`. This works in flat `node_modules` layouts (npm, yarn) because `@types/react` is hoisted from the consuming project. When pnpm hoisting is disabled (`hoist: false`), TypeScript cannot find the React types and fails with:

```
error TS2307: Cannot find module 'react' or its corresponding type declarations.
```

**Workaround:**
```yaml
packageExtensions:
  "@xyflow/react@12":
    peerDependencies:
      "@types/react": "*"
```

**Suggested fix:** Add `@types/react` to `peerDependencies` in `@xyflow/react`'s `package.json`.

--- a/.changeset/react-types-peer-dep.md
+++ b/.changeset/react-types-peer-dep.md
@@ -1,0 +1,5 @@
+---
+'@xyflow/react': patch
+---
+
+Add `@types/react` and `@types/react-dom` as optional peer dependencies so the package resolves React types under pnpm strict mode (`hoist: false`) without forcing JavaScript consumers to install the type packages.

--- a/.changeset/react-types-peer-dep.md
+++ b/.changeset/react-types-peer-dep.md
@@ -2,4 +2,4 @@
 '@xyflow/react': patch
 ---
 
-Add `@types/react` and `@types/react-dom` as optional peer dependencies so the package resolves React types under pnpm strict mode (`hoist: false`) without forcing JavaScript consumers to install the type packages.
+Add `@types/react` and `@types/react-dom` as optional peer dependencies to prevent issues with pnpm strict mode (`hoist: false`)

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -66,8 +66,18 @@
     "zustand": "^4.4.0"
   },
   "peerDependencies": {
+    "@types/react": ">=17",
+    "@types/react-dom": ">=17",
     "react": ">=17",
     "react-dom": ">=17"
+  },
+  "peerDependenciesMeta": {
+    "@types/react": {
+      "optional": true
+    },
+    "@types/react-dom": {
+      "optional": true
+    }
   },
   "devDependencies": {
     "@types/node": "^18.7.16",

